Fix Aircraft handling of resupply/Enter orders

reaperrr opened the pull request:
Fixes #16784.

reaperrr committed:
Improve AircraftCanEnter readability

reaperrr committed:
Fix Repairable returning order targeter for aircraft

Aircraft (currently) does its own order targeting for resupplies,
so this could lead to conflicts.

reaperrr committed:
Fix aircraft allowing enter when it should not

Show "enter-blocked" cursor when resupply is not possible,
except when
-  the actor CanForceLand,
- does not TakeOffOnResupply,
- and has active ForceEnter modifier.

pchote commented:
Planes now correct show the "cannot enter" cursor, but will still happily land on and block the service depot when ordered.

reaperrr commented:
I feared as much, but I'm a bit at a loss on how to fix that. I can't just perform the same checks I use for the cursor for the "can target" check, because then the `enter(-blocked)` cursor won't show at all. How do we implement this case - generally targetable, but blocked - normally?

pchote commented:
I suspect we issue the order but then don't act on it from `ResolveOrder` - but you should double check how other cases handle this.

reaperrr commented:
Updated.

pchote commented on OpenRA.Mods.Common/Traits/Air/Aircraft.cs:
```diff
@@ -996,19 +1012,24 @@ public void ResolveOrder(Actor self, Order order)
 
+				var targetActor = order.Target.Actor;
+				var isForceEnter = orderString == "ForceEnter";
+				var canResupplyAt = isForceEnter ? AircraftCanResupplyAt(targetActor, !Info.TakeOffOnResupply) : AircraftCanResupplyAt(targetActor);
```
This can simplify to `var canResupplyAt = AircraftCanResupplyAt(targetActor, isForceEnter && !Info.TakeOffOnResupply);`

pchote reviewed (dismissed):
The changes here LGTM aside from one minor cleanup.

While testing I noticed that helicopters go to the service depot rally point as expected, but planes do not. I'm pretty sure that is a different bug, so can be deferred to a different PR if you don't want to fix it here.

reaperrr commented:
Updated.

> While testing I noticed that helicopters go to the service depot rally point as expected, but planes do not. I'm pretty sure that is a different bug, so can be deferred to a different PR if you don't want to fix it here.

As discussed on IRC, that's the result of a work-around to prevent aircraft without `TakeOffOnResupply` from blocking other actors until we've refactored the docking/reservation logic to be agnostic of the resupply type, so out of scope for this release cycle.
